Lib/textwrap.py

```diff
@@ -428,7 +428,8 @@ def dedent(text):
     # all lines.
     margin = None
     text = _whitespace_only_re.sub('', text)
-    indents = _leading_whitespace_re.findall(text)
+    indents = [line[:-len(line.lstrip())] for line in text.splitlines() if
```

Comment:
line.lstrip() sems to be called twice here.

An alternative could be to replace both the list comprehension and the loop over the resulting list by a straight-forward loop like (untested):

```
    for line in text.splitlines():
        lstr = line.lstrip()
        if not lstr:
            continue
        indent = line[:-len(lstr)]
```

Reply:
This seems to work, taking all the comments into account leads to the following diff:

-    indents = [line[:-len(line.lstrip())] for line in text.splitlines() if
-               line.lstrip()]
-    for indent in indents:
+    split_text = text.splitlines(True)
+    for line in split_text:
+        lstr = line.lstrip()
+        if not lstr:
+            continue
+        indent = line[:-len(lstr)]

and 

-        line[margin_length:] if line.lstrip()
-        else line[len(line.splitlines()[0]):]
-        for line in text.splitlines(True)
+        line[margin_length if line.lstrip() else len(line.splitlines()[0]):]
+        for line in split_text

Reply:
Hey @SylvainDe Thanks for your review :)

I was thinking why do we want to check for `if not lstr` and then ultimately do a `continue`. IMHO, we can just do `if lstr`. Also this makes me think why are we not storing the indents in an array? 

The local variable indent would keep on updating with every iteration of the loop.
Am I missing something here?

Reply:
Hello,

1. That's a good question. I usually never use `continue` so I am really surprised I introduced it (and went to the original code to be sure) and don't understand the reason. I'd strongly prefer a version without the continue :)

2. The `indent` variable is different at each iteration - just like in the removed code `for indent in indents:`. Is that a problem ?

I hope my answers are not too silly or imprecise - to be fair, I don't remember much of the original code/your commit/my suggestion and code is hard to read/test on my current device. Please let me know if you need a more precise answer - I'll be happy to try if required.

Reply:
We would still need the indent list here. I've made the changes taking into account your review about calculating `line.lstrip()` twice by converting the list comprehension to a straightforward loop.

Let me know if this looks now :)